include/serenity.proto

```diff
@@ -0,0 +1,129 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+* Proto for serenity framework. Describes Input and the Output of the whole flow
+* and internal messages
+**/
+
+//Because serenity is a separate project, you have to copy mesos.proto to same directory
+import "mesos.proto";
+
+package mesos;
+
+option java_package = "org.apache.mesos";
+option java_outer_classname = "Protos";
+
+/**
+* NOTE: These proto buffers should be upstreamed - it defines
+* external communication (slave -> serenity -> slave)
+**/
+
+/**
+* Contains resource usage for running executors
+**/
+message Usage {
+ repeated ResourceUsage resource_usage = 1;
+}
+
+
+/**
+* Contains slack resources which can be offered for BE tasks
+**/
+message Slack {
+  repeated Resource resources = 1;
+}
+
+
+/**
+* Describes the type of correction.
+**/
+enum CorrectionAction {
+  KILL = 1;
+  FREEZE = 2;
+  RESIZE = 3;
```

Comment:
I thought of throttling by using Resize Action + Labels that contains metadata for resize correction action (like resize: [[cpu, 2.0]]). Future Resize-complaint QoS could initiate throttling based on these observations.